input/tar2/tar_input.go

```diff
@@ -0,0 +1,163 @@
+package tar2
+
+import (
+	"archive/tar"
+	"bytes"
+	"crypto/sha512"
+	"encoding/base64"
+	"hash"
+	"io"
+	"os"
+	"path"
```

Comment:
You're right, [`Clean` doesn't handle dirseps](https://play.golang.org/p/-L_Ca9hgTI).

... I guess that's not wrong.  '\' is a valid part of a filename on unix.  Heck, it is on windows too; if you talk to nfts directly without explorer, it'll let you do all sorts of things (or at least, my fuse-ntfs mount on linux is letting me `mkdir \\`, which is a little scary).  So `..\..` might *look* funny, but it is actually a valid filename, and we should accept it.

A build of the CLI components (serving as an uploader, without the containment components) on windows might want to see things differently... but we'll cross that bridge when we come to it, I guess.